### PR TITLE
Support “Pending” test case Tag #839

### PR DESCRIPTION
**Pull request by jnd-au**
- Also show elapsed time for JS/Native Ignored tests (may be non-zero), to be consistent with JVM

### CODE REVIEW

**Review by tgodzik — CHANGES_REQUESTED** [review]
Sounds reasonable to add it. Sorry for taking so long to review!

File: junit-interface/src/main/java/munit/internal/junitinterface/EventDispatcher.java [hunk 1]
```diff
@@ -133,10 +134,29 @@ public void testIgnored(Description desc) {
         desc,
         new InfoEvent(desc, Status.Ignored) {
           void logTo(RichLogger logger) {
+            StringBuilder builder = new StringBuilder(1024);
+            boolean isPending = false;
+            for (Annotation annotation : desc.getAnnotations()) {
+              if (annotation instanceof Tag) {
+                Tag tag = (Tag) annotation;
+                String kind = tag.getClass().getName();
+                if (kind.equals("munit.Tag") && tag.value().equals("Pending")) {
```
[the remark this answers is not in the document]
Comment (jnd-au) [reply junit-interface/src/main/java/munit/internal/junitinterface/EventDispatcher.java:143]: Thanks for your review! The reason for lines 143-146 was for simplicity, to avoid adding lots of Java and JS interfaces in the `munit.internal.junitinterface` package. However, based on your feedback, I’ll push some changes to that interface.

File: munit/shared/src/main/scala/munit/package.scala [hunk 2]
```diff
@@ -1,7 +1,10 @@
 package object munit {
+  case class PendingComment(override val value: String) extends Tag(value)
```
[the remark this answers is not in the document]
Comment (jnd-au) [reply munit/shared/src/main/scala/munit/package.scala:2]: Well, we need to differentiate the special sentinel-value flag from the arbitrary user-comment instances, so I felt that the cleanest way is with types, not string values. If we use a special string value like you suggested, then tags would be coalesced together and then we need to separate them out again, with separate implementations for Java and ScalaJS. Here’s is what the implementations might look like:

```java
              boolean isPending = false;
              ...
                if (tag instanceof PendingTag) {
                  if (tag.value().equals("Pending")) {
                    isPending = true;
                  }
                  else {
                    builder.append(" ");
                    builder.append(tag.value());
                  }
                }
                ...
            if (isPending) {
              builder.insert(0, " PENDING");
            }
```

```scala
      val (isPending, pendingComments) = annotations.collect {
        case pending: PendingTag => pending
      }.partition(_.equals(munit.Pending))
      isPending.distinct.map(_.value.toUpperCase) ++ pendingComments.map(_.value)
```

This is okay, but somehow these implementations, and their equivalency, seem more unclear to me. So I didn’t push a commit for this.

File: munit/shared/src/main/scala/munit/MUnitRunner.scala [hunk 3]
```diff
@@ -325,7 +325,7 @@ class MUnitRunner(val cls: Class[_ <: Suite], newInstance: () => Suite)
       catch onError
     result.map { _ =>
       notifier.fireTestFinished(description)
-      true
+      !test.tags(Pending)
```
[the remark this answers is not in the document]
Comment (jnd-au) [reply munit/shared/src/main/scala/munit/MUnitRunner.scala:328]: Yes, PendingComment uses Pending (because `def pending(comment: String): TestOptions = pending.tag(PendingComment(comment))`), so we only need to check for Pending.
